src/main/java/com/google/sps/servlets/NotificationServlet.java

```diff
@@ -0,0 +1,195 @@
+        Notification notification = new Notification();
+        notification.message = resultSet.getString(SqlConstants.NOTIFICATION_FETCH_MESSAGE);
+        notification.url = resultSet.getString(SqlConstants.NOTIFICATION_FETCH_URL);
+        notification.timestamp = resultSet.getTimestamp(SqlConstants.NOTIFICATION_FETCH_TIMESTAMP);
+        // Store object in ArrayList.
+        notifications.add(notification);
+      }
+      connection.close();
+    } catch (SQLException ex) {
+        Logger logger = Logger.getLogger(NotificationServlet.class.getName());
+        logger.log(Level.SEVERE, ex.getMessage(), ex);
+    }
+    return notifications;
+  }
+
+  /**
+   * Receives the data for a notification and inserts it into the Notification table.
+   */
+  private void insertToNotification(Connection connection, String message, String notificationUrl,
+        Timestamp dateTime) {
+    String query = "INSERT INTO Notification(message, url, date_time) VALUES(?,?,?)";
+    try {
+      // Prepare the statement to be inserted.
+      PreparedStatement prepStatement = connection.prepareStatement(query);
+      prepStatement.setString(SqlConstants.NOTIFICATION_INSERT_MESSAGE, message);
+      prepStatement.setString(SqlConstants.NOTIFICATION_INSERT_URL, notificationUrl);
+      prepStatement.setTimestamp(SqlConstants.NOTIFICATION_INSERT_DATETIME, dateTime);
+      prepStatement.executeUpdate();
+    } catch (SQLException ex) {
+      Logger logger = Logger.getLogger(NotificationServlet.class.getName());
+      logger.log(Level.SEVERE, ex.getMessage(), ex);
+    }
+  }
+
+  /**
+   * Receives the data for a notification-user relation and inserts it into
+   * UserNotification.
+   */
+  private void insertToUserNotification(Connection connection, int userId, int notificationId) {
+    String query = "INSERT INTO UserNotification(user_id, notification_id) VALUES(?,?)";
+    try {
+      // Prepare the statement to be inserted.
+      PreparedStatement prepStatement = connection.prepareStatement(query);
+      prepStatement.setInt(SqlConstants.USER_NOTIFICATION_INSERT_USERID, userId);
+      prepStatement.setInt(SqlConstants.USER_NOTIFICATION_INSERT_NOTIFICATIONID, notificationId);
+      prepStatement.executeUpdate();
+    } catch (SQLException ex) {
+      Logger logger = Logger.getLogger(NotificationServlet.class.getName());
+      logger.log(Level.SEVERE, ex.getMessage(), ex);
+    }
+  }
+
+  /**
+   * Returns the ID of a just inserted notification.
+   */
+  private int getLastInsertedNotificationId(Connection connection) {
+    String query =  "SELECT id FROM Notification ORDER BY date_time";
+    // Query the information from Notification table.
+    int notificationId = 0;
+    try (PreparedStatement pst = connection.prepareStatement(query);
+         ResultSet resultSet = pst.executeQuery()) {
+      // Select the id of the last notification to be inserted.
+      resultSet.last();
+      notificationId = resultSet.getInt(1);
```

Comment:
nit: can you make this '1' into a constant?

Reply:
Done!